File: core/src/main/java/com/ataulm/basic/Id.java
package com.ataulm.basic;

import com.google.auto.value.AutoValue;

import java.util.UUID;

@AutoValue
public abstract class Id {

    public static Id create() {
        UUID uuid = UUID.randomUUID();
        return new AutoValue_Id(uuid);
    }

    public static Id createFrom(String rawUuid) {
        UUID uuid = UUID.fromString(rawUuid);
        return new AutoValue_Id(uuid);
    }

    public abstract UUID value();

}

File: core/src/main/java/com/ataulm/chunks/Id.kt
package com.ataulm.chunks

import java.util.*

data class Id(val value: UUID) {

    companion object {

        fun create(): Id {
            val uuid = UUID.randomUUID()
            return Id(uuid)
        }

        fun createFrom(rawUuid: String): Id {
            val uuid = UUID.fromString(rawUuid)
            return Id(uuid)
        }
    }

}

